automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/azure/AzureDiskSpec.scala

```diff
@@ -0,0 +1,189 @@
+            .disk(
+              new AzureDiskConfig()
+                .name(diskName)
+                .size(50)
+                .labels(labelMap)
+            )
+
+          _ <- IO(runtimeClient.createAzureRuntime(workspaceId, runtimeName.asString, false, createReq))
+          _ <- loggerIO.info(s"AzureDiskSpec: Create runtime request submitted. Starting to poll GET")
+
+          // Verify the initial getRuntime call
+          _ <- IO.sleep(5 seconds)
+          callGetRuntime = IO(runtimeClient.getAzureRuntime(workspaceId, runtimeName.asString))
+
+          intitialGetRuntimeResponse <- callGetRuntime
+          _ <- loggerIO.info(s"initial get runtime response ${intitialGetRuntimeResponse}")
+          _ = intitialGetRuntimeResponse.getStatus shouldBe ClusterStatus.CREATING
+
+          _ <- loggerIO.info(
+            s"AzureDiskSpec: runtime ${workspaceId}/${runtimeName.asString} in creating status detected"
+          )
+
+          _ <- loggerIO.info("AzureDiskSpec: verifying get disk response")
+          diskId = intitialGetRuntimeResponse.getRuntimeConfig.getAzureConfig.getPersistentDiskId
+          getDisk = IO(diskClient.getDiskV2(diskId.toBigInteger.intValue()))
+          diskDuringRuntimeCreate <- getDisk
+          _ = diskDuringRuntimeCreate.getStatus shouldBe DiskStatus.CREATING
+
+          _ <- loggerIO.info(
+            s"AzureDiskSpec: disk ${workspaceId}/${diskDuringRuntimeCreate.getId()} in creating status detected"
+          )
+
+          // Verify the runtime eventually becomes Running (in 40 minutes)
+          monitorCreateResult <- streamUntilDoneOrTimeout(
+            callGetRuntime,
+            240,
+            10 seconds,
+            s"AzureDiskSpec: runtime ${workspaceId}/${runtimeName.asString} did not finish creating after 20 minutes"
+          )(implicitly, GeneratedLeonardoClient.runtimeInStateOrError(ClusterStatus.RUNNING))
+
+          _ <- loggerIO.info(
+            s"AzureDiskSpec: runtime ${workspaceId}/${runtimeName.asString} create monitor result: $monitorCreateResult"
+          )
+          _ = monitorCreateResult.getStatus() shouldBe ClusterStatus.RUNNING
+
+          _ <- loggerIO.info(
+            s"AzureDiskSpec: runtime ${workspaceId}/${runtimeName.asString} delete starting"
+          )
+
+          // Delete the runtime but not the disk
+          _ <- IO(runtimeClient.deleteAzureRuntime(workspaceId, runtimeName.asString, false))
+
+          _ <- loggerIO.info(
+            s"AzureDiskSpecAzureDiskSpec: runtime ${workspaceId}/${runtimeName.asString} delete called"
+          )
+
+          // Wait until disk is unattached
+          // TODO: is there a better way to do this?
+          monitorGetRuntimeUntilUnattached <- streamUntilDoneOrTimeout(
+            callGetRuntime,
+            240,
+            10 seconds,
+            // TODO: increase timeout
+            s"AzureDiskSpec: disk ${workspaceId}/${diskName} was not ready after 5 minutes"
```

Comment:
5 -> 40?

Reply:
ah yeah, a few of these were copy pasta clearly 😅